pkg/testing/fixture.go

```diff
@@ -709,26 +709,21 @@ func (e *ExecErr) Unwrap() error {
 // ExecStatus executes the status subcommand on the prepared Elastic Agent binary.
 // It returns the parsed output and the error from the execution. Keep in mind
 // the agent exits with status 1 if it's unhealthy, but it still outputs the
-// status successfully. Therefore, a non-empty AgentStatusOutput is valid
-// regardless of the error. An empty AgentStatusOutput and non nil error
-// means the output could not be parsed. Use AgentStatusOutput.IsZero() to
-// determine if the returned AgentStatusOutput is empty or not.
+// status successfully. An empty AgentStatusOutput and non nil error
+// means the output could not be parsed.
+// As long as we get some output, we don't return any error.
 // It should work with any 8.6+ agent
 func (f *Fixture) ExecStatus(ctx context.Context, opts ...process.CmdOption) (AgentStatusOutput, error) {
-	out, err := f.Exec(ctx, []string{"status", "--output", "json"}, opts...)
+	out, _ := f.Exec(ctx, []string{"status", "--output", "json"}, opts...)
+
 	status := AgentStatusOutput{}
 	if uerr := json.Unmarshal(out, &status); uerr != nil {
 		return AgentStatusOutput{},
-			fmt.Errorf("could not unmarshal agent status output: %w",
-				errors.Join(&ExecErr{
-					err:    err,
-					Output: out,
-				}, uerr))
+			fmt.Errorf("could not unmarshal agent status output: %w", uerr)
+	} else if status.IsZero() {
+		return status, fmt.Errorf("agent status output is empty")
```

Comment:
I would keep the error and log it here in case it isn't empty, for example file not found or some permissions issue on the CI runner.

Reply:
I didn't get you. Can you please elaborate?

Reply:
You are throwing away the `error` return value from `out, _ := f.Exec(ctx, []string{"status", "--output", "json"}, opts...)`. The logic you have is correct to ignore it for deciding whether the command succeeded, but if the command failed and status is empty we should still wrap the original error in case it contains helpful context for why the command failed.

I want this to be `return status, fmt.Errorf("agent status output is empty: %w", err)` instead.

Reply:
thanks craig.